models/repo.go

```diff
@@ -135,6 +136,13 @@ func NewRepoContext() {
 	RemoveAllWithNotice("Clean up repository temporary data", filepath.Join(setting.AppDataPath, "tmp"))
 }
 
+func init() {
+	services.RegisterService("models", func() error {
+		NewRepoContext()
+		return nil
+	}, "setting")
+}
+
```

Comment:
A lot of these service functions are written like this - shouldn't NewRepoContext be changed to be `func() error`?

Reply:
Moved more but kept some of them.